Utility/Framework/SymfonyFramework.php

```diff
@@ -75,19 +78,6 @@ public static function isKernelLessThan($major, $minor = null, $patch = null)
      */
     private static function kernelVersionCompare($operator, $major, $minor = null, $patch = null)
     {
-        $vernum = $major;
-        $kernel = Kernel::MAJOR_VERSION;
-
-        if ($minor) {
-            $vernum .= '.'.$minor;
-            $kernel .= '.'.Kernel::MINOR_VERSION;
-
-            if ($patch) {
-                $vernum .= '.'.$patch;
-                $kernel .= '.'.Kernel::RELEASE_VERSION;
-            }
-        }
-
-        return version_compare($kernel, $vernum, $operator);
+        return version_compare(Kernel::VERSION_ID, sprintf("%d%'.02d%'.02d", $major, $minor ?: 0, $patch ?: 0), $operator);
```

Comment:
By the way, I found that the `:?` is actually not required, as `null` is cast to `0` automatically, but I found this made the usage more clear, and I don't like relying on PHP's internal type casting.